lightdash/lightdash
# fix: cartesian chart config resetting on table calculation update

Relates to: [#7996](https://github.com/lightdash/lightdash/issues/7996)

### Description:

- Fixes updating table calculation resetting chart config
- Fixes being able to change cartesian chart type
- Adds e2e test to make sure that updating a TC doesn't change the chart config
- Fixes e2e test of changing chart type to wait for the stores to fully update - the second bug that went to production was that whenever we changed the chart type it went back to the initial one after briefly changing so the test passed even though it was incorrect

*Steps to test*:
1. Create a chart that uses a table calculation
2. Try to change chart from bar to other type (line, etc)
3. Assign table calculation to one of the axis
4. Check that updating the table calculation doesn't change the chart configuration

https://github.com/lightdash/lightdash/assets/22939015/aa1c1f1f-5a60-4845-8c52-a90ebf9060ac

### Reviewer actions
